modules/openapi-generator/src/main/resources/python/python-experimental/model_utils.mustache

```diff
@@ -609,6 +612,35 @@ def deserialize_model(model_data, model_class, path_to_item, check_type,
         instance = used_model_class(**kw_args)
     return instance
 
+recursive_discriminators_cache = {}
+
+def recursive_discriminators(model_class):
+    """
+    Return the discriminator mapping for all possible subtypes (recursively found down to the leaves)
+    :param model_class:
+    :return:
+    """
+    if model_class in recursive_discriminators_cache:
+        return recursive_discriminators_cache[model_class]
+    recursive_discriminated_types = model_class.discriminator()
+    if not recursive_discriminated_types:
+        return {}
```

Comment:
Our old invocation of get_discriminator_class returns a class.
Shouldn't we return model_class in this case?

Reply:
This isn't equivalent to get_discriminator_class for performance reasons. This instead is equivalent to model_class.discriminator.

Reply:
Why do we need to store an empty dict if we will never key into it?
Shouldn't we avoid making that last recursive call?